feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -849,7 +852,11 @@ class BillingWrapper(
                     // only setObfuscatedAccountId for non-upgrade/downgrades until google issue is fixed:
                     // https://issuetracker.google.com/issues/155005449
                     replaceSkuInfo?.let {
-                        setUpgradeInfo(it)
+                        setSubscriptionUpdateParams(
+                            SubscriptionUpdateParams.newBuilder()
+                            .setOldPurchaseToken(replaceSkuInfo.oldPurchase.purchaseToken)
+                            .setReplaceProrationMode(ProrationMode.IMMEDIATE_AND_CHARGE_FULL_PRICE)
```

Comment:
this is the main change -- how we pass upgrade info into the purchase flow. 

i defaulted to IMMEDIATE_AND_CHARGE_FULL_PRICE for testing purposes, but we should pass through the one passed to the method.

one note: defaulting to null didn't seem to work, and there are still some open questions here. if you see any weird behavior, please document, but otherwise i think we can split proration mode work out to a separate ticket